kleinanzeigen_bot/extract.py

```diff
@@ -105,10 +105,40 @@ def extract_shipping_info_from_ad_page(self) -> (str, float | None):
                 shipping_price = float(parse_decimal(shipping_price_parts[-2]))
                 ship_type = 'SHIPPING'
                 ship_costs = shipping_price
+
+                # extract shipping options
+                # It is only possible the extract the cheapest shipping option,
+                # as the other options are not shown
+                shipping_option_mapping = {
+                    "DHL_2": "5,49",
+                    "Hermes_Päckchen": "4,50",
+                    "Hermes_S": "4,95",
+                    "DHL_5": "6,99",
+                    "Hermes_M": "5,95",
+                    "DHL_10": "9,49",
+                    "DHL_31,5": "16,49",
+                    "Hermes_L": "10,95",
+                }
+                for shipping_option, shipping_price in shipping_option_mapping.items():
+                    if shipping_price in shipping_text:
+                        shipping_options = [shipping_option]
+                        break
         except NoSuchElementException:  # no pricing box -> no shipping given
             ship_type = 'NOT_APPLICABLE'
 
-        return ship_type, ship_costs
+        return ship_type, ship_costs, shipping_options
+
+    def extract_sell_directly_from_ad_page(self) -> bool | None:
+        """
+        Extracts the sell directly option from an ad page.
+
+        :return: a boolean indicating whether the sell directly option is active (optional)
+        """
+        try:
+            buy_now_is_active = self.webdriver.find_element(By.ID, 'j-buy-now').text == "Direkt kaufen"
+            return buy_now_is_active
+        except NoSuchElementException:
+            return None
```

Comment:
It is either Direct Selling active or not, right? Imo it would make a bit more sense to just return False here instead